### ISSUE TITLE
Alt text popover does not work as expected on hidden media

### ISSUE THREAD

**Issue by nclm** [mastodon/mastodon#32172]
### Steps to reproduce the problem

1. Find a media that is hidden and has alt text (or hide a media that has alt text)
2. Click on the “ALT” badge to read the alt text (feature introduced in #32058)

### Expected behaviour

The alt text popover from #32058 should open to read the alt text

### Actual behaviour

The media is unhidden, and the alt text popup is not opened

### Mastodon instance

mastodon.social

### Mastodon version

v4.3.0-beta.2+pr-32093-e7137df

### Browser name and version

Firefox 131.0 (aarch64)

### Operating system

macOS 14.3 (23D56)

**Comment by nclm** [comment]
As a quick test, moving `media-gallery__item__badges` out of `media-gallery__item` (as a sibling of `spoiler-button__overlay`) and adding a high `z-index` to it seems to make it work as expected.

**Comment by diondiondion** [comment]
Closing this as the Alt badge is not displayed on hidden media anymore.
